src/main/java/ip/Parser.java

```diff
@@ -0,0 +1,75 @@
+package ip;
+
+import java.util.Scanner;
+
+import ip.command.AddCommand;
+import ip.command.ByeCommand;
+import ip.command.Command;
+import ip.command.EditCommand;
+import ip.command.FindCommand;
+import ip.command.ListCommand;
+import ip.exception.InvalidCommand;
+
+/**
+ * Parses input from the user.
+ *
+ * @author Jonathan Lam
+ */
+public class Parser {
+    /** String that the user has entered */
+    private Scanner inputLine;
+
+    /**
+     * Stores user input string as a Scanner object.
+     *
+     * @param s Scanner to be stored.
+     */
+    public void load(Scanner s) {
+        inputLine = s;
```

Comment:
Might be better to have it as Scanner inputLine so this line written as inputLine = inputLine